src/components/radio-group/radio-group.ts

```diff
@@ -93,12 +87,20 @@ export default class IgcRadioGroupComponent extends LitElement {
   }
 
   private _observerCallback() {
-    const radios = Array.from(this._radios as unknown as Element[]).filter(
-      (el) => el.tagName.toLowerCase() === 'igc-radio'
-    ) as IgcRadioComponent[];
-    this.disabled = radios.every((radio) => radio.disabled);
-    this.labelBefore =
-      radios.some((radio) => radio.labelPosition === 'before') ?? false;
+    const radios = Array.from(this._radios);
+    const setState = (state: string, condition: boolean) =>
+      condition
+        ? this._internals.states.add(state)
+        : this._internals.states.delete(state);
+
+    setState(
+      'disabled',
+      radios.every((radio) => radio.disabled)
+    );
+    setState(
+      'labelBefore',
```

Comment:
```suggestion
      'label-before',
```
IMO it would be better to stick to the native CSS pseudo selectors/elements schema, which is usually kebab cased; `nth-child`, `first-of-type`, etc.